File: guessing-game-bdd-atomic/src/test/java/tw/waterballsa/bddworkshop/stepdefs/ScenarioContext.java
package tw.waterballsa.bddworkshop.stepdefs;

import org.springframework.stereotype.Component;
import org.springframework.test.web.servlet.MvcResult;

import java.util.HashMap;
import java.util.Map;

@Component
public class ScenarioContext {
    
    private MvcResult lastResponse;
    private Map<String, Object> contextData = new HashMap<>();
    
    public MvcResult getLastResponse() {
        return lastResponse;
    }
    
    public void setLastResponse(MvcResult lastResponse) {
        this.lastResponse = lastResponse;
    }
    
    public void clear() {
        this.lastResponse = null;
    }
    
    public void reset() {
        this.lastResponse = null;
        this.contextData.clear();
    }
    
    public void put(String key, String value) {
        contextData.put(key, value);
    }
    
    public Object get(String key) {
        return contextData.get(key);
    }
} 

File: guessing-game-bdd-atomic/features/support/scenario.context.ts
import { Response } from 'supertest';

export class ScenarioContext {
  private lastResponse: Response;
  private contextData: Map<string, any> = new Map();

  getLastResponse(): Response {
    return this.lastResponse;
  }

  setLastResponse(response: Response): void {
    this.lastResponse = response;
  }

  clear(): void {
    this.lastResponse = null;
  }

  reset(): void {
    this.lastResponse = null;
    this.contextData.clear();
  }

  put(key: string, value: any): void {
    this.contextData.set(key, value);
  }

  get(key: string): any {
    return this.contextData.get(key);
  }
}
